content/en/tracing/trace_collection/automatic_instrumentation/dd_libraries/go.md

```diff
@@ -31,17 +39,163 @@ The Go Tracer requires Go `1.18+` and Datadog Agent `>= 5.21.1`. For a full list
 
 For configuration instructions and details about using the API, see the Datadog [API documentation][4].
+{{% /collapse-content %}}
+
+{{% collapse-content title="Automatic compile-time instrumentation (alpha)" level="p" %}}
+
+<div class="card callout-card mb-4">
+  <div class="card-body d-flex flex-column">
+    <h5 class="card-title text-black mt-0 mb-1">Orchestrion is currently in alpha</h5>
+    <p class="card-text">
+      Orchestrion is under active development, and Datadog is eager to hear from you. Request access if you are interested and would like to share your experience.
+    </p>
+    <a href="TBD"
+      target="_blank"
+      class="btn btn-outline-primary pb-1 align-self-end d-flex flex-column justify-content-center"
+    >
+      Request Access
+    </a>
+  </div>
+</div>
+
+### Overview
+
+[Orchestrion][6] is Datadog's automatic compile-time instrumentation tool for Go applications. It automatically adds instrumentation (backed by the [Go Datadog Tracing library][4]) to applications during compilation.
+
+Key benefits:
+
+- Instruments your code and all dependencies, including the Go standard library
+- Provides the best possible tracing coverage
+- Enables certain [Application Security Management][7] features exclusive to Orchestrion-built Go applications
+
+Running during compilation allows `orchestrion` to not only instrument your own code, but also all of your dependencies' code, including the Go standard library, ensuring you get the best tracing coverage possible.
+
+### Requirements
+
+- Supports the two latest Go runtime releases (matching [Go's official release policy][8]). Older Go runtimes may function, but are not supported by default. For special requests, [contact support][9].
+- Applications must be managed using [go modules][10]. Module vendoring is supported.
+
+#### Supported packages
+
+Orchestrion is under active development and supports a subset of integrations available in the [tracing library][1]. For the latest list of supported frameworks and their minimum Orchestrion versions, see the [Supported frameworks](#supported-frameworks).
+
+### Install Orchestrion
+
+To install and set up Orchestrion:
+
+1. Install Orchestrion:
+   ```sh
+   go install github.com/datadog/orchestrion@latest
+   ```
+   <div class="alert alert-info">Ensure <code>$(go env GOBIN)</code> (or <code>$(go env GOPATH)/bin</code>) is in your <code>$PATH</code>.</div>
+   
+1. Register Orchestrion in your project's `go.mod`:
+   ```sh
+   orchestrion pin
+   ```
+1. Commit changes to your version control system:
+   ```sh
+   git commit -m "chore: enable orchestrion" go.mod go.sum orchestrion.tool.go
+   ```
+
+Now you can manage your dependency on `orchestrion` like any other dependency using the `go.mod` file.
+
+### Usage
+
+Use one of these methods to enable Orchestrion in your build process:
```

Comment:
Do we prefer any of these methods over the other, or are there any scenarios that a user would prefer one over the other? We could potentially add those details to help a user pick. Alternatively, we could just show one option as the happy path.

Reply:
My personal (and absolutely un-biased opinion as the lead engineer for Orchestrion) opinion is that Orchestrion should be the golden path going forward; however since the current state of it is "Alpha", I think it would be weird to present it as the recommended option right now...

And since we're making these doc edits to try and attract more early adopters, I'd also prefer to avoid presenting the manual approach as "preferred" :)

Reply:
Agreed! I suppose I was asking more about the three ways to enable Orchestrion. We could potentially be more opinionated in this section or list why a user would choose one of the three methods over the other.